### PR TITLE
create and delete all account integrations

### PR DESCRIPTION
**Pull request by Varsha-Arun**
Creating and deleting all account integrations. Completed account integration test suite in 47secs

selection_999 1301

selection_999 1302

### CODE REVIEW

**Comment by Varsha-Arun** [comment]
selection_999 1323

latest PR(https://github.com/Shippable/bvt/pull/2/commits/e65df7a09f1fca4f6703d1c179154fdfcc5e9f4a)  testing subscription API calls for 
1. GET subscriptions for owner, member, collaborator and public user
2. GET subscription by ID for owner, member, collaborator, unauthorized and public user
3. GET run status by subscription ID for owner, member, collaborator, unauthorized and public user

File: tests/core/api/GH_ORG_SUBS_API.js [hunk 1]
```diff
@@ -0,0 +1,436 @@
+'use strict';
```
Comment (avinci) [inline tests/core/api/GH_ORG_SUBS_API.js]: the only issue i have here is the name of the file. its not GH_ORG...its just subscriptions.js right? The api does not give you different results when you use different source control. by qualifying this by GH_ORG you are hinting that we should do a bitbucket test also which is pointless.. All we are testing here is the subs and making sure the subs are returned correctly.

File: tests/core/api/GH_ORG_SUBS_API.js [hunk 2]
```diff
@@ -0,0 +1,436 @@
...
+              global.ADM_GH_SUB_COUNT, 'Subscription count needs to match');
+            return done();
+          }
+        );
+      }
+    );
+
+    it('2. Owner can get subscription by ID',
+      function (done) {
+        var sub = _.findWhere(subscriptions, {orgName:global.TEST_GH_ORGNAME}) || {};
```
Comment (avinci) [inline tests/core/api/GH_ORG_SUBS_API.js]: why do we need to do this? why not just use the project that you setup, use its subscriptionId

File: tests/core/api/GH_ORG_SUBS_API.js [hunk 3]
```diff
@@ -0,0 +1,436 @@
...
+            assert.isNotEmpty(subscription, 'User cannot find the subscription by ID');
+            return done();
+          }
+        );
+      }
+    );
+
+    it('3. Member can get subscription by ID',
+      function (done) {
+        var sub = _.findWhere(subscriptions, {orgName:global.TEST_GH_ORGNAME}) || {};
```
Comment (avinci) [inline tests/core/api/GH_ORG_SUBS_API.js]: same issue here, we should use the same project and make sure all the folks can get it.

File: tests/core/api/GH_ORG_SUBS_API.js [hunk 4]
```diff
@@ -0,0 +1,436 @@
...
+            assert.isNotEmpty(subscription, 'User cannot find the subscription by ID');
+            return done();
+          }
+        );
+      }
+    );
+
+    it('5. Public user cannot get subscription by ID',
+      function (done) {
+        var sub = _.findWhere(subscriptions, {orgName:global.TEST_GH_ORGNAME}) || {};
```
Comment (avinci) [inline tests/core/api/GH_ORG_SUBS_API.js]: we are missing one more test here. We should that public users cannot get any values if the make a getS call without Ids. We should also test getS call by passing a subscriptionId

File: tests/core/api/GH_ORG_SUBS_API.js [hunk 5]
```diff
@@ -0,0 +1,436 @@
...
+              util.format('User should not be able to get the runstatus by subscription ID. ' +
+                'err : %s %s', err, runStas)
+            );
+            return done();
+          }
+        );
+      }
+    );
+
+    it('16. public user cannot get Run Status by subscription ID',
```
Comment (avinci) [inline tests/core/api/GH_ORG_SUBS_API.js]: we also need to test the public project where the unauthorized and public guys can get this information correct?

**Comment by Varsha-Arun** [comment]
Updated the PR
Added tests for GET build status from subscriptions/{subscriptionId}
And whether it is private or public project response for getting latest runStatus through subscription ID is same for unauthorized and public user.

**Comment by Varsha-Arun** [comment]
Adding project API call tests

selection_999 1333

- GET projects
- GET project by ID
- POST triggerNewBuild by project ID

**Pull request merged**